zokrates_core/src/proof_system/bn128/utils/parser.rs

```diff
@@ -0,0 +1,43 @@
+use std::collections::HashMap;
+
+pub fn parse_vk(vk: String) -> HashMap<String, String> {
+    let mut reader = csv::ReaderBuilder::new()
+        .delimiter(b'=')
+        .has_headers(false)
+        .trim(csv::Trim::All)
+        .from_reader(vk.as_bytes());
+
+    let mut map = HashMap::new();
+    let mut iterator = reader.deserialize::<(String, String)>();
+
+    while let Some(r) = iterator.next() {
+        if r.is_ok() {
+            let r = r.unwrap();
+            map.insert(r.0, r.1);
+        }
+    }
+    map
+}
+
+#[cfg(test)]
+mod tests {
+    use proof_system::bn128::utils::parser::parse_vk;
+
+    #[test]
```

Comment:
this needs more negative tests with invalid files

Reply:
Added 2 more tests